crates/ruma-common/src/identifiers/room_or_alias_id.rs

```diff
@@ -11,7 +11,8 @@ use super::{server_name::ServerName, OwnedRoomAliasId, OwnedRoomId, RoomAliasId,
 ///
 /// `RoomOrAliasId` is useful for APIs that accept either kind of room identifier. It is converted
 /// from a string slice, and can be converted back into a string as needed. When converted from a
-/// string slice, the variant is determined by the leading sigil character.
+/// string slice, the variant is determined by the leading sigil character. It can also be converted
+/// to an `RoomId` or `RoomAliasId` by using its `TryFrom` or `TryInto` implementations.
```

Comment:
```suggestion
/// to a `RoomId` or `RoomAliasId` by using its `TryFrom` or `TryInto` implementations.
```

Also can you put this in a new paragraph? And I believe we should make it clearer that the user can `match` to get either.